airbyte-integrations/connectors/source-snowflake/src/main/java/io.airbyte.integrations.source.snowflake/SnowflakeDataSourceUtils.java

```diff
@@ -37,8 +38,9 @@ public class SnowflakeDataSourceUtils {
   public static final String AIRBYTE_OSS = "airbyte_oss";
   public static final String AIRBYTE_CLOUD = "airbyte_cloud";
   private static final String JDBC_CONNECTION_STRING =
-      "role=%s&warehouse=%s&database=%s&schema=%s&JDBC_QUERY_RESULT_FORMAT=%s&CLIENT_SESSION_KEEP_ALIVE=%s&application=%s";
+      "role=%s&warehouse=%s&database=%s&JDBC_QUERY_RESULT_FORMAT=%s&CLIENT_SESSION_KEEP_ALIVE=%s&application=%s";
 
+  private static final String JDBC_SCHEMA_PARAM = "&schema=%s&CLIENT_METADATA_REQUEST_USE_CONNECTION_CTX=true";
```

Comment:
is the problem with the current implementation that it does not set `CLIENT_METADATA_REQUEST_USE_CONNECTION_CTX` to `true` and therefore connector ends up pulling all tables from all schemas?

Reply:
Correct. 
So while UI is requiring a schema field, it doesn't affect the outcome catalog